### ISSUE TITLE
Feature Request - edit source code in an indirect buffer like `org-src-edit-mode`

### ISSUE THREAD

**Issue by peterhoeg** [leon-barrett/just-mode.el#8]
In an org-mode file, it's possible to inject a source block written in any language supported by emacs and then switch to the corresponding major mode when editing that source block which will give you proper syntax highlighting, flymake/lsp support and so on - basically all the trimmings of the regular major mode, which is *very* nice.

As a `Justfile` can also contain scripts, it would be very neat if we were able to do something similar. org-mode knows which major mode to switch to based on the `#+BEGIN_SRC` tag and so just-mode would have to do something clever about the shebang line.

**Comment by leon-barrett** [comment]
I agree, that would be nice. I'm not prepared to write sufficiently advanced elisp right now, sorry. I'm open to contributions.

**Comment by leon-barrett** [comment]
A workaround has been merged in https://github.com/leon-barrett/just-mode.el/pull/14 that allows editing of just that recipe via `C-c '`. That may help, but it's not the full major-mode-embedded-in-a-recipe logic that I think you were asking for.
